low-level/framework/messaging/egress_accumulated_msgs_processor.py

```diff
@@ -124,14 +127,24 @@ def run(self):
                     dict_msg = json.loads(message)
                     if dict_msg.get("iem"):
                         try:
+                            # check if IEM Framework initialized,
+                            # if not, retry initializing the IEM Frameowork
+                            if os.path.exists(IEM_INIT_FAILED):
+                                with open(IEM_INIT_FAILED, 'r') as f:
+                                    sspl_pid = f.read()
+                                if sspl_pid and psutil.pid_exists(int(sspl_pid)):
+                                    EventMessage.init(component='sspl', source='S')
+                                    logger.info(
+                                        "IEM framework initialization completed!!")
+                                os.remove(IEM_INIT_FAILED)
                             EventMessage.send(
                                 module=dict_msg["iem"]["module"],
                                 event_id=dict_msg["iem"]["event_code"],
                                 severity=dict_msg["iem"]["severity"],
                                 message_blob=dict_msg["iem"]["description"])
                             logger.info("Accumulated IEM sent.")
                             self.store_queue.delete()
```

Comment:
Hi @shriya-deshmukh , I don't have much knowledge about IEM framework, but we have IEM wrapper class in sspl right? so can we write and use a common function to send the EventMessage from that class, then the same function can handle such cases like EventMessage is initialized or not, etc..

Reply:
Done, added common function in Iem class to send IEM msg.

Reply:
That's great 😀